functions/vshn-postgres-func/encrypted_pvc_test.go

```diff
@@ -0,0 +1,79 @@
+package vshnpostgres
+
+import (
+	"context"
+	"testing"
+
+	xkube "github.com/crossplane-contrib/provider-kubernetes/apis/object/v1alpha1"
+	"github.com/go-logr/logr"
+	"github.com/stretchr/testify/assert"
+	vshnv1 "github.com/vshn/component-appcat/apis/vshn/v1"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/yaml"
+)
+
+func TestNoEncryptedPVC(t *testing.T) {
+	type args struct {
+		expectedFuncIO string
+		inputFuncIO    string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "GivenNoEncryptionParams_ThenExpectNoOutput",
+			args: args{
+				expectedFuncIO: "enc_pvc/01-ThenExpectNoOutput.yaml",
+				inputFuncIO:    "enc_pvc/01-GivenNoEncryptionParams.yaml",
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+			log := logr.FromContextOrDiscard(ctx)
+
+			iof := getFunctionFromFile(t, tt.args.inputFuncIO)
+			comp := &vshnv1.VSHNPostgreSQL{}
+			inComp := getCompositeFromIO(t, iof, comp)
+			expIof := getFunctionFromFile(t, tt.args.expectedFuncIO)
+
+			_, err := AddUserAlerting(ctx, log, iof, inComp)
+
+			if tt.wantErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, expIof, iof)
+			}
+
+		})
+	}
+}
+
+func TestGivenEncrypedPvcThenExpectOutput(t *testing.T) {
+
+	ctx := context.Background()
+	log := logr.FromContextOrDiscard(ctx)
+
+	t.Run("GivenEncryptionEnabled_ThenExpectOutput", func(t *testing.T) {
+
+		iof := getFunctionFromFile(t, "enc_pvc/02-GivenEncryptionParams.yaml")
+		comp := &vshnv1.VSHNPostgreSQL{}
+		inComp := getCompositeFromIO(t, iof, comp)
+
+		_, err := AddPvcSecret(ctx, log, iof, inComp)
+		assert.NoError(t, err)
+
+		resName := inComp.Name + "-luks-key"
+		kubeObject := &xkube.Object{}
+		assert.NoError(t, iof.GetManagedRessourceFromDesired(resName, kubeObject))
+		s := &v1.Secret{}
+		assert.NoError(t, yaml.Unmarshal(kubeObject.Spec.ForProvider.Manifest.Raw, s))
+		assert.NotEmpty(t, s.Data["luksKey"])
+	})
+
+}
```

Comment:
Please add at least a testcase with `enabled: false`. That way you'd have caught the bug

Reply:
Done